### PR TITLE
Updated for TPP 5.2 release version

### PR DESCRIPTION
**Pull request by edeutsch**
# Submitting a Container
 
 (If you're requesting for a new container, please check the procedure described [here](https://github.com/BioContainers/containers#241-how-to-request-a-container).  
 
 ## Check BioContainers' Dockerfile [specifications](https://github.com/BioContainers/specs)  
 ## Checklist
 
 1. Misc
 - [ ] My tool doesn't exist in [BioConda](#make-sure-your-tool-isnt-already-present-in-bioconda)   
 - [ ] The image can be built   
 
 2. [Metadata](#check-biocontainers-dockerfile-metadata)  
 - [ ] LABEL base_image  
 - [ ] LABEL version
 - [ ] LABEL software.version  
 - [ ] LABEL about.summary  
 - [ ] LABEL about.home  
 - [ ] LABEL about.license   
 - [ ] MAINTAINER  
 
 3. Extra (optionals)
  - [ ] I have written tests in test-cmds.txt  
  - [ ] LABEL extra.identifier    
  - [ ] LABEL about.documentation   
  - [ ] LABEL about.license_file
  - [ ] LABEL about.tags   
 
 
 ## Check [BioContainers'](https://github.com/BioContainers/specs) Dockerfile metadata

### CODE REVIEW

File: tpp/5.0/Dockerfile [hunk 1]
```diff
@@ -1,34 +1,39 @@
 ################## BASE IMAGE ######################
-FROM biocontainers/biocontainers:v1.0.0_cv4
```
Comment (osallou) [inline tpp/5.0/Dockerfile]: Please specify a base image tag

File: tpp/5.0/Dockerfile [hunk 2]
```diff
@@ -1,34 +1,39 @@
...
+FROM biocontainers/biocontainers
 
 ################## METADATA ######################
-LABEL base_image="biocontainers:v1.0.0_cv4"
-LABEL version="4"
+LABEL base_image="biocontainers"
+LABEL version="5.2"
```
Comment (osallou) [inline tpp/5.0/Dockerfile]: This is a new release, it should be in its own directory tpp/5.2/Dockerfile

In a version dir, only bug related fixes on some version are allowed

**Review by osallou — CHANGES_REQUESTED** [review]

**Review by osallou — APPROVED** [review]

**Comment by osallou** [comment]
Having some issues with our CI, will merge after fix

Thanks for your work

**Comment by osallou** [comment]
Hi CI system is back and found an error.
software name label must be your directory name and software must be in lowercase.
So you should update software.name to tpp instead of TPP
Thanks

**Comment by edeutsch** [comment]
Thanks for the response, but I don't think I really understand. I tried to follow the TPP 5.0 template quite closely. Is that one valid or also invalid?
There is no label "software.name". TPP 5.0 didn't have it. I just checked PeptideShaker as another example and it doesn't have it. I can add it, but I wonder if that is useful? There is just a plain label "software". And that is capital TPP. But it was also caps in 5.0 and 4.8. And for PeptideShaker, there are also caps. So is that really the problem? Are all the others invalid, too?

**Comment by osallou** [comment]
Found an existing bioconda package for this software (https://bioconda.github.io/recipes/tpp/README.html), is this the same, then you should update the recipe in bioconda to avoid duplicates.

**Comment by edeutsch** [comment]
I updated the label "software" from TPP to tpp in a new pull request.
Have never dealt with bioconda so I don't know what to do with that. That is TPP 5.0, much older than 5.2.

**Comment by osallou** [comment]
No tpp/5.2/test-cmds.txt (test file) present, skipping tests

**Comment by edeutsch** [comment]
I'm happy to create a test-cmds.txt file. Are there any other examples that I can work from? I don't see another single test-cmds.txt file in the entire repo. Am I the guinea pig for a whole new CI system? These issues don't seem to apply to any other project in the repo?

**Comment by osallou** [comment]
Don't worry about last comments, those are auto generated and only recommendations generated for each build.
Goal is to have checks passed. Other comments are ways to improve quality

**Comment by osallou** [comment]
Found an existing bioconda package for this software (https://bioconda.github.io/recipes/tpp/README.html), is this the same, then you should update the recipe in bioconda to avoid duplicates.

**Comment by edeutsch** [comment]
ah, I see. Some of the comments above that appear to written by you are in fact written by Jenkins. Is there actually interest in having a test-cmds.txt file? I can write a super simple one that can perform a basic little test. Although getting it to work properly in your framework is probably something that you'll need to polish.

**Comment by osallou** [comment]
No tpp/5.2/test-cmds.txt (test file) present, skipping tests

**Comment by edeutsch** [comment]
Well, since I have it up on the screen here, the following is something that could be used to test the container at a very basic level. You'll need to figure out what works with your framework:

mkdir /tmp/tppdata
chmod 777 /tmp/tppdata
docker run -v /tmp/tppdata:/data biocontainers/tpp comet -p
if [ -f /tmp/tppdata/comet.params.new ]; then
    /bin/rm -rf /tmp/tppdata
    exit 0
fi
/bin/rm -rf /tmp/tppdata
echo "ERROR: Comet did not generate comet.params.new"
exit 10

**Pull request merged**